### PR TITLE
Implement a signoff request button in the builtin signoff plugin

### PR DESCRIPTION
**Pull request by n1k0**
This is WiP. This enhances the plugin API to allow:

- Registering plugins, so the plugin can be aware of the store (and so of the `dispatch` function, useful to dispatch actions)
- Registering plugin component hooks, allowing to augment pieces of UI for a given page container component.

### CODE REVIEW

File: scripts/store/configureStore.js [hunk 1]
```diff
@@ -14,10 +14,12 @@ const finalCreateStore = compose(
 )(createStore);
 
 export default function configureStore(initialState, plugins=[]) {
-  const store = finalCreateStore(createRootReducer(plugins), initialState);
+  const pluginReducers = plugins.map(({reducers={}}) => reducers);
```
Comment (n1k0) [inline scripts/store/configureStore.js]: XXX comment heavily

**Comment by n1k0** [comment]
Deployed to gh-pages. Deploy command used:

```
$ KINTO_ADMIN_PLUGINS=signoff npm run publish-to-gh-pages
```

File: plugins/signoff/index.js [hunk 2]
```diff
@@ -1,29 +1,87 @@
...
+// Actions
+function requestSignoff() {
+  return {type: PLUGIN_SIGNOFF_REQUEST};
+}
+
+
+function* handleSignoffRequest(getState, action) {
+  // Obtain current bucket and collection ids from state.
+  const {collection:collectionState} = getState();
+  const {bucket: bid, name: cid} = collectionState;
```
Comment (Natim) [inline plugins/signoff/index.js]: why do you put spaces between : and bid and not between : and collectionState?

```
const {collection: collectionState} = getState();
const {bucket: bid, name: cid} = collectionState;
```
```
const {collection:collectionState} = getState();
const {bucket:bid, name:cid} = collectionState;
```

File: plugins/signoff/index.js [hunk 3]
```diff
@@ -1,29 +1,87 @@
...
+
+    // Hide button if server does not support signoff.
+    const capability = serverInfo.capabilities.signer;
+    if (!capability) {
+      return null;
+    }
+
+    // Hide button if this collection is not configured to be signed.
+    const sources = capability.resources.map((r) => `${r.source.bucket}/${r.source.collection}`);
+    if (!sources.includes(`${bid}/${cid}`)) {
```
Comment (Natim) [inline plugins/signoff/index.js]: :+1:

**Comment by Natim** [comment]
This is awesome thanks. I wish to make it available in the [kinto-admin server plugin](https://github.com/mozilla-services/kinto-dist/blob/master/Makefile#L39-L42).

**Pull request merged**